mdsuite/calculators/green_kubo_distinct_diffusion_coefficients.py

```diff
@@ -152,15 +164,18 @@ def run_experimental_analysis(self):
         if type(self.atom_selection) is dict:
             select_atoms = {}
             for item in self.atom_selection:
-                select_atoms[str.encode(join_path(item, "Velocities"))] = self.atom_selection[item]
+                select_atoms[str.encode(join_path(item, "Velocities"))] = \
+                self.atom_selection[item]
             self.atom_selection = select_atoms
         for combination in self.combinations:
             type_spec = {}
             data_path = [join_path(item, 'Velocities') for item in combination]
             self._prepare_managers(data_path=data_path)
             type_spec = self._update_species_type_dict(type_spec, data_path, 3)
-            type_spec[str.encode('data_size')] = tf.TensorSpec(None, dtype=tf.int16)
-            batch_generator, batch_generator_args = self.data_manager.batch_generator(dictionary=True)
+            type_spec[str.encode('data_size')] = tf.TensorSpec(None,
+                                                               dtype=tf.int16)
```

Comment:
we should start using `self.dtype` - but here we are really going for int16? I don't thing int32 would be slower but safer longterm?

Reply:
absolutely. I will update that.